src/property.hpp

```diff
@@ -139,10 +140,8 @@ namespace realm {
                 return "array";
             case PropertyType::LinkingObjects:
                 return "linking objects";
-#if __GNUC__
```

Comment:
This was deliberately only including the default: case for GCC and not Clang because without it clang can warn that the switch needs to be updated if another case is added to the enum, but GCC warns about not all cases being covered even when they are. If VC++ also warns without this change then it should just be checking for GCC or VC++.